src/MinFraud.php

```diff
@@ -113,82 +113,366 @@ public function __construct(
-            $values = Util::maybeHashEmail($values);
+        $new = $this;
+        if (\array_key_exists('account', $values)) {
+            $new = $new->withAccount($this->remove($values, 'account', ['array']));
+        }
+        if (\array_key_exists('billing', $values)) {
+            $new = $new->withBilling($this->remove($values, 'billing', ['array']));
+        }
+        if (\array_key_exists('credit_card', $values)) {
+            $new = $new->withCreditCard($this->remove($values, 'credit_card', ['array']));
+        }
+        if (\array_key_exists('custom_inputs', $values)) {
+            $new = $new->withCustomInputs($this->remove($values, 'custom_inputs', ['array']));
+        }
+        if (\array_key_exists('device', $values)) {
+            $new = $new->withDevice($this->remove($values, 'device', ['array']));
+        }
+        if (\array_key_exists('email', $values)) {
+            $new = $new->withEmail($this->remove($values, 'email', ['array']));
+        }
+        if (\array_key_exists('event', $values)) {
+            $new = $new->withEvent($this->remove($values, 'event', ['array']));
+        }
+        if (\array_key_exists('order', $values)) {
+            $new = $new->withOrder($this->remove($values, 'order', ['array']));
+        }
+        if (\array_key_exists('payment', $values)) {
+            $new = $new->withPayment($this->remove($values, 'payment', ['array']));
+        }
+        if (\array_key_exists('shipping', $values)) {
+            $new = $new->withShipping($this->remove($values, 'shipping', ['array']));
+        }
+        if (\array_key_exists('shopping_cart', $values)) {
+            foreach ($this->remove($values, 'shopping_cart', ['array']) as $item) {
+                $new = $new->withShoppingCartItem($item);
+            }
         }
 
-        $new = clone $this;
-        $new->content = $values;
+        $this->verifyEmpty($values);
 
         return $new;
     }
 
     /**
      * This returns a `MinFraud` object with the `device` array set to
-     * `$values`. Existing `device` data will be replaced.
+     * to the values provided. Existing `device` data will be replaced.
+     *
+     * @param array       $values         An array of device data. The keys are the same as
+     *                                    the JSON keys. You may use either this or the named
+     *                                    arguments, but not both.
+     * @param string|null $ipAddress      The IP address associated with the device
+     *                                    used by the customer in the transaction.
+     *                                    The IP address must be in IPv4 or IPv6
+     *                                    presentation format, i.e., dotted-quad
+     *                                    notation or the IPv6 hexadecimal-colon
+     *                                    notation.
+     * @param string|null $userAgent      the HTTP `User-Agent` header of the browser
```

Comment:
`s/the HTTP/The HTTP/`. A couple other descriptions also start with a lower case letter